sdks/python/apache_beam/typehints/schemas.py

```diff
@@ -205,6 +218,11 @@ def typing_from_runner_api(fieldtype_proto):
     pass  # TODO
 
 
+def _hydrate_namedtuple_instance(encoded_schema, values):
+  return named_tuple_from_schema(
+      proto_utils.parse_Bytes(encoded_schema, schema_pb2.Schema))(*values)
+
+
 def named_tuple_from_schema(schema):
```

Comment:
Does it makes sense to memoize this?

Reply:
It's effectively memoized with SCHEMA_REGISTRY inside `typing_from_runner_api`. We could short-circuit it here as well though

Reply:
I went ahead and merged this, let me know if you think this should be tweaked and I can do it separately.